### ISSUE TITLE
Automatic updates via rack middleware

### ISSUE THREAD

**Issue by dnamsons** [railsware/js-routes#288]
Hi, thanks for the great library!

I recently updated this library in one of my projects and while doing so I also migrated it to webpacker. 
While doing so, I noticed that this library currently performs auto updates via `js.erb` and therefore doesn't support auto generation for TypeScript definitions. 

I wanted to suggest an alternative route for performing auto updates - a custom rack middleware like in [i18n-js](https://github.com/fnando/i18n-js/blob/main/lib/i18n/js/middleware.rb). This would allow for automatic updates for TypeScript definitions as well. 

I have already made such a middleware, so this issue is more about trying to find out if upstreaming something like this is desirable.

**Comment by bogdan** [comment]
Implemented in version 2.2.0. Recommended setup described in [Readme](https://github.com/railsware/js-routes#quick-start). Feel free to follow up if any questions or bugs.

**Comment by eelkevdbos** [comment]
First of all, thank you for this is a wonderful feature. I just have one question with regards to the default behavior. Are the `routes.js` and `routes.d.ts` files automatically generated during `asset:compile` for the non-development environments or do we have to commit the generated artifacts generated by the development middleware?

It could be very convenient to have both files generated in the asset pipeline for the other environments so feature branch workflows do not have to spin up the application and do a recompile of the files or otherwise resolving the merge manually.

If this is already default behavior, consider this question not asked :)

**Comment by bogdan** [comment]
That is good point. That is not default behaviour and I don't feel like it can be because it is hard to turn it off once it is done.

After quick googling I found that the best way to enhance `assets:precompile` is via the following inside `Rakefile`:

``` ruby
namespace :assets do
  task :precompile => "js:routes:typescript"
end
```

I also think it is a good idea to add those files to `.gitignore` by default if we are absolutely sure most deployment processes call `assets:precompile` on server side as part of deployment process.
Overall someone may still have problems in having those files available and up to date if one would work only inside test environment or console, but I think we can live with that.

I don't manage a deployment for any rails application at the moment, so can you tell me if this variant works best for you?

**Comment by bogdan** [comment]
I've implemented all of that in `v2.2.1`. Follow up if any feedback or open a new issue.
